app/views/promotion/newspaperLandingPage.scala.html

```diff
@@ -7,20 +7,20 @@
 @import model.Subscriptions._
 
 @(catalog: Catalog, promoCode: PromoCode, promotion: PromoWithNewspaperLandingPage, md: MarkdownRenderer)
-@voucherPlans = @{catalog.voucher.list.filter(p => promotion.appliesTo.productRatePlanIds.contains(p.id))}
-@deliveryPlans = @{catalog.delivery.list.filter(p => promotion.appliesTo.productRatePlanIds.contains(p.id))}
+@voucherPlans = @{catalog.voucher.list.toList.filter(p => promotion.appliesTo.productRatePlanIds.contains(p.id))}
+@deliveryPlans = @{catalog.delivery.list.toList.filter(p => promotion.appliesTo.productRatePlanIds.contains(p.id))}
 @anyPromotion = @{asAnyPromotion(promotion)}
 @title = @{"Subscribe to the Guardian and Observer"}
-@hash = @{s"${promotion.landingPage.defaultProduct}"}
+@hash = @{s"${promotion.landingPage.value.defaultProduct}"}
 @products = @{
     val adapter = planToOptions(promoCode, anyPromotion) _
     val voucherProduct = CollectionSubscriptionProduct(
         title = "",
         description = "",
         altPackagePath = "#delivery",
         options = {
-            voucherPlans.filter(_.charges.digipack.isEmpty).map(adapter).sortBy(_.weeklyPrice).reverse ++
-            voucherPlans.filter(_.charges.digipack.isDefined).map(adapter).sortBy(_.weeklyPrice).reverse
+            voucherPlans.filter(_.charges.digipack.isEmpty).map(adapter).sortBy(_.weeklyPrice).reverse.toList ++
```

Comment:
I think the .toList is not necessary here as it's already been converted in line 10

Reply:
even intellij told me that too! well spotted
![image](https://user-images.githubusercontent.com/7304387/35985170-e4bdc02c-0ced-11e8-9907-0b6235238f9e.png)